reactor-core/build.gradle

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2011-2018 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+if (JavaVersion.current().java9Compatible) {
+  sourceSets {
+	java8.java.srcDirs = ['src/main/java8']
```

Comment:
I'd merge this `sourceSet` (and the corresponding directory) into `java9`

Reply:
`java9` source set is included into the final JAR, but classes from `java8` must be not.
Also, it depends on `main` source set. But these classes have to be on compile classpath of `main`. If we add `compileOnly java9`, it will create a cycle

Reply:
with the renaming of the directory to `java8stubs`, I'm good with it 👍